src/backend/PortabilityService.Gateway/Reliability/PortabilityServiceQoSProvider.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Ocelot.Configuration;
+using Ocelot.Requester.QoS;
+using Polly;
+using Polly.Timeout;
+
+namespace PortabilityService.Gateway.Reliability
+{
+    /// <summary>
+    /// Custom quality-of-service provider to create Polly policies tailored
+    /// to PortabilityService scenarios.
+    /// 
+    /// Note that these policies are currently very similar to Ocelot defaults 
+    /// because there's not a lot of flexibility in the types of policies Ocelot accepts. 
+    /// There is a work item (https://github.com/ThreeMammals/Ocelot/issues/264) open to fix 
+    /// that and this type should be updated according to the comments below once that's done.
+    /// </summary>
+    public class PortabilityServiceQoSProvider : IQoSProvider
+    {
+        private readonly ILogger _logger;
+        private readonly CircuitBreaker _circuitBreaker;
+
+        public PortabilityServiceQoSProvider(DownstreamReRoute reRoute, ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory.CreateLogger<PortabilityServiceQoSProvider>();
```

Comment:
Can you just ask for a `ILogger<PortabilityServiceQoSProvider>` in the constructor arg?

Reply:
Unfortunately we can't because `PortabilityServiceQoSProviderFactory` is what's registered with DI and it instantiates instances of this type. So, we get the `ILoggerFactory` when the factory is created and then pass it along to the type to make its own logger.